### PR TITLE
As an API user, I can click `forgot password` button and receive the link via email

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Add  forgot password API
 
## Insight
- Reuse `PasswordController` from Devise.
- Skip `verify_authenticity_token` and override `respond_with` method to act as an API
- Need to tell Devise which mapping to use in RSpec
 
## Proof Of Work
[Postman collection](https://nimblehq.postman.co/collections/11835486-63105f38-c20f-493d-b4d8-873a18c8d8c7?workspace=9daf6b25-882e-4ad0-8299-95529e2883ff)

### CODE REVIEW

File: app/controllers/api/v1/forgot_passwords_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class ForgotPasswordsController < ApplicationController
```
Comment (ankitkalia1195) [inline app/controllers/api/v1/forgot_passwords_controller.rb:5]: Can we just name it passwordsController, to follow naming of devise?

File: app/controllers/api/v1/forgot_passwords_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      skip_before_action :doorkeeper_authorize!
+
+      before_action :set_resource
+
+      def create
```
Comment (ankitkalia1195) [inline app/controllers/api/v1/forgot_passwords_controller.rb:10]: @hoangmirs I think we might not need controller at all, can we try using endpoint already provided by devise?

Comment (hoangmirs) [reply app/controllers/api/v1/forgot_passwords_controller.rb]: Thank you, I change to inherit from PasswordController of Devise and just change a bit to response JSON data

File: config/routes.rb [hunk 3]
```diff
@@ -11,4 +11,12 @@
       end
     end
   end
+
+  namespace :api do
+    namespace :v1 do
+      devise_scope :user do
+        post 'forgot_password', to: 'passwords#create'
```
Comment (ankitkalia1195) [inline config/routes.rb:18]: route can also be named just `/passwords`

File: spec/controllers/api/v1/forgot_passwords_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
```
Comment (ankitkalia1195) [inline spec/controllers/api/v1/forgot_passwords_controller_spec.rb:2]: @hoangmirs We can rename the file here to match class name

File: app/controllers/api/v1/passwords_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+      protected
+
+      def respond_with(*resource)
+        user = resource.first
+
+        if user.blank?
+          render json: {}, status: :created
+        else
+          render json: { error: user.errors.full_messages.to_sentence }, status: :not_found
```
Comment (olivierobert) [inline app/controllers/api/v1/passwords_controller.rb:16]: Returning `not_found` actually exposes us to user enumeration as an attacker can know what email exists. So it's usually better to always return success with a message like "If an account matched the email provided, a reset email has been sent". I think that's the default behaviour of Devise.

**Pull request merged**